WPMedia/fusion-news-theme-blocks
# TBRANDS-81 -- Update list and sales price display logic

## Description
On the Product Assortment Carousel when list and sale price are displayed, a third condition needed to be added to the list price and sale price logic.

If there is a list price and sale price but they are the same, the item isn't on sale so don't display a sale price, just list price.


## Jira Ticket

- [TBRANDS-81](https://arcpublishing.atlassian.net/browse/TBRANDS-81)

## Acceptance Criteria

Desired State:

```
if price_list is available, display price_list. (no change)

if price_list and price_sale is available, display price_sale, and display price_list with a strikethrough. (no change)

if price_list and price_sale are the same value: only display price_list.
```

## Test Steps
1. Checkout this branch `git checkout TBRANDS-81--pricing-logic`
2. Run storybook with `npm run storybook`
3. View Product Assortment Carousel in storybook, and scroll to the sixth item in the carousel (data was added to the algolia mock data)
4. Confirm only list price displays and sale price does not.


If you want to then compare it to the base branch for existing functionality you'd need repeat the above steps except on `arc-themes-release-version-2.0.1` branch and add the following test case to `algoliaProduct.js` before running storybook:
```
	{
		image: testImageURL,
		sku: "sku-5",
		prices: [
			{
				amount: 26,
				type: "List",
				currencyLocale: "en-US",
				currencyCode: "USD",
			},
			{
				amount: 26,
				type: "Sale",
				currencyLocale: "en-US",
				currencyCode: "USD",
			},
		],
	},
```

## Effect Of Changes
### Note:
My storybook isn't showing the strikethrough styles, shouldn't impact this PR's logic as it's unrelated but calling it out when  see the before/after screenshots. Not sure if there is a regression or it's my local.

### Before
Sale and list price display
![image](https://user-images.githubusercontent.com/2506649/181374225-5729b5a3-0227-4d15-80e3-8a8326ac9418.png)

### After
List price only displays
![image](https://user-images.githubusercontent.com/2506649/181372937-0a4128a2-3492-4208-9363-324a002c53d5.png)


## Dependencies or Side Effects
None I'm aware of.

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
